### PR TITLE
combine ESRestorePillowCheckpoints with PillowCheckpointSeqStore

### PR DESCRIPTION
**Pull request by snopoke**
This change combined these two models which were almost identical. It also gives us a bigger window over which we would get notifications about rewinds and be able to fix them.

### CODE REVIEW

File: corehq/apps/hqadmin/tasks.py [hunk 1]
```diff
@@ -3,18 +3,31 @@
...
+        has_rewound, historical_seq = check_for_rewind(checkpoint)
+        if has_rewound:
+            notify_exception(
+                None,
+                message='Found seq number lower than previous for {}. '
+                        'This could mean we are in a rewind state'.format(checkpoint.checkpoint_id),
+                details={
+                    'pillow checkpoint seq': checkpoint.get_current_sequence_id(),
+                    'stored seq': historical_seq
+                }
```
Comment (dimagimon) [inline corehq/apps/hqadmin/tasks.py]: E122 continuation line missing indentation or outdented

File: corehq/apps/hqadmin/models.py [hunk 2]
```diff
@@ -60,24 +74,21 @@ def get_by_pillow_name(cls, pillow_name):
...
-                               checkpoint_id=checkpoint.checkpoint_id,
-                               date_updated=date.today())
+    def get_historical_max(cls, checkpoint_id):
+        try:
+            return cls.objects.filter(checkpoint_id=checkpoint_id).order_by('-seq_int')[0]
+        except IndexError:
+            return None
+
+    class Meta:
+        ordering = ['-date_updated']
```
Comment (benrudolph) [inline corehq/apps/hqadmin/models.py]: 👍 attention to detail

**Review by benrudolph — APPROVED** [review]
nice cleanup

**Comment by dannyroberts** [comment]
> Conflicting files
> corehq/apps/hqadmin/management/commands/fix_checkpoint_after_rewind.py

**Pull request merged**